tests/integration/test_create_token.py

```diff
@@ -0,0 +1,138 @@
+                },
+                "receipting_keys": ["qid"],
+            },
+            "collection_exercise_sid": "789",
+            "response_id": "1234567890123456",
+            "language_code": "en",
+            "roles": [],
+            "account_service_url": ACCOUNT_SERVICE_URL,
+        },
+    },
+]
+
+"""
+The purpose of this test class is to test the creation of a token (from create_token.py) to ensure metadata in
+a decrypted token is nested/found in the correct level. create_token_v2 is used as v1 methods are intended to
+be dropped and the other tests only use methods that do not have a v2 implementation.
+"""
+
+
+class TestCreateToken(IntegrationTestCase, AppContextTestCase):
+    def test_payload_from_token(self):
+        for value in test_parameters:
+            with self.subTest():
+                additional_payload = value.get("additional_payload")
+                token = self.token_generator.create_token_v2(
+                    schema_name=value.get("schema"),
+                    theme=value.get("theme"),
+                    **additional_payload,
+                )
+                with self.test_app.app_context():
+                    decrypted_token = decrypt_token(token)
+                    self.assertEqual(
+                        decrypted_token, value.get("payload") | decrypted_token
+                    )
+                    assert (
+                        additional_payload.items()
+                        <= decrypted_token.get("survey_metadata").get("data").items()
+                    )
+
+    def test_uuid_consistent_after_decryption(self):
+        token = self.token_generator.create_token_v2(
+            "test_checkbox.json", theme="social", value="Dummy Text"
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            assert decrypted_token.get("survey_metadata") == {
+                "data": {
+                    "case_ref": "1000000000000001",
+                    "qid": PAYLOAD_V2_SOCIAL.get("survey_metadata")
+                    .get("data")
+                    .get("qid"),
+                    "value": "Dummy Text",
+                },
+                "receipting_keys": ["qid"],
+            }
+
+    def test_supplementary_data_included_in_token(self):
+        token = self.token_generator.create_supplementary_data_token(
+            "test_checkbox.json", flag_1=True
+        )
+        with self.test_app.app_context():
+            decrypted_token = decrypt_token(token)
+            self.assertEqual(
+                decrypted_token, PAYLOAD_V2_SUPPLEMENTARY_DATA | decrypted_token
+            )
+            assert ({"flag_1": True}).items() <= decrypted_token.get(
+                "survey_metadata"
+            ).get("data").items()
+
+    def test_metadata_is_removed_from_token(self):
+        for value in [
+            [self.token_generator.create_token_without_jti, "jti"],
```

Comment:
Could make this a dict to make it more readable in the test below?

Reply:
Updated